test_src/sljitTest.c

```diff
@@ -11813,34 +11844,45 @@ static void test92(void)
 #endif /* SLJIT_64BIT_ARCHITECTURE */
 	FAILED(buf[19] != -573621, "test92 case 26 failed\n");
 	FAILED(*(sljit_u8*)(buf + 20) != 240, "test92 case 27 failed\n");
-	FAILED(((sljit_u8*)(buf + 20))[1] != 0x55, "test92 case 28 failed\n");
+	FAILED(((sljit_u8*)(buf + 20))[1] != PADBYTE, "test92 case 28 failed\n");
 	FAILED(buf[21] != 192, "test92 case 29 failed\n");
 	FAILED(buf[22] != -5893, "test92 case 30 failed\n");
 	FAILED(buf[23] != 4059, "test92 case 31 failed\n");
 	FAILED(buf[24] != 6359, "test92 case 32 failed\n");
 	FAILED(buf[25] != (sljit_sw)(buf + 23), "test92 case 33 failed\n");
-	FAILED(((sljit_u8*)(buf + 26))[0] != 0x55, "test92 case 34 failed\n");
+#if (!defined SLJIT_ATOMIC_MIN_WIDTH || SLJIT_ATOMIC_MIN_WIDTH <= 8)
+	FAILED(((sljit_u8*)(buf + 26))[0] != PADBYTE, "test92 case 34 failed\n");
 	FAILED(((sljit_u8*)(buf + 26))[1] != 204, "test92 case 35 failed\n");
-	FAILED(((sljit_u8*)(buf + 26))[2] != 0x55, "test92 case 36 failed\n");
+	FAILED(((sljit_u8*)(buf + 26))[2] != PADBYTE, "test92 case 36 failed\n");
 	FAILED(buf[27] != 105, "test92 case 37 failed\n");
-	FAILED(((sljit_u8*)(buf + 28))[1] != 0x55, "test92 case 38 failed\n");
+	FAILED(((sljit_u8*)(buf + 28))[1] != PADBYTE, "test92 case 38 failed\n");
 	FAILED(((sljit_u8*)(buf + 28))[2] != 240, "test92 case 39 failed\n");
-	FAILED(((sljit_u8*)(buf + 28))[3] != 0x55, "test92 case 40 failed\n");
+	FAILED(((sljit_u8*)(buf + 28))[3] != PADBYTE, "test92 case 40 failed\n");
 	FAILED(buf[29] != 13, "test92 case 41 failed\n");
 	FAILED(buf[30] != 0, "test92 case 42 failed\n");
+#endif /* SLJIT_ATOMIC_MIN_WIDTH <= 8 */
+#if (!defined SLJIT_ATOMIC_MIN_WIDTH || SLJIT_ATOMIC_MIN_WIDTH <= 16)
 	FAILED(((sljit_u16*)(buf + 31))[0] != 0x5555, "test92 case 43 failed\n");
 	FAILED(((sljit_u16*)(buf + 31))[1] != 51403, "test92 case 44 failed\n");
 	FAILED(buf[32] != 14876, "test92 case 45 failed\n");
+#endif /* SLJIT_ATOMIC_MIN_WIDTH <= 16 */
 #if (defined SLJIT_64BIT_ARCHITECTURE && SLJIT_64BIT_ARCHITECTURE)
+#if PADBYTE != 0
 	FAILED(((sljit_u32*)(buf + 33))[0] != 0x55555555, "test92 case 46 failed\n");
+#endif
 	FAILED(((sljit_u32*)(buf + 33))[1] != 0xdeadbeef, "test92 case 47 failed\n");
-	FAILED(buf[34] != 0xffffffff, "test92 case 48 failed\n");
+	FAILED(((sljit_u32*)(buf + 34))[0] != 0xffffffff, "test92 case 48 failed\n");
 #endif /* SLJIT_64BIT_ARCHITECTURE */
 	FAILED(buf[35] != 1, "test92 case 49 failed\n");
+#if (!defined SLJIT_ATOMIC_MIN_WIDTH || SLJIT_ATOMIC_MIN_WIDTH <= 8)
 	FAILED(buf[36] != 0x11, "test92 case 50 (load) failed\n");
 	FAILED(((sljit_u8*)(buf + 37))[1] != buf[36], "test92 case 50 (store) failed\n");
+#endif /* SLJIT_ATOMIC_MIN_WIDTH <= 8 */
```

Comment:
this test will need rewriting once we better understand the constrains that NOT supporting byte addressable/aligned bytes in hardware causes.

when "emulation" is enabled to allow code that can pack atomics in bytes like this test does, I would expect failures, and indeed have a version that failed so often that I added a loop to it with 8 attempts and that is not included.

I also implemented similar code using 32bit and 64bit atomics (that are supported natively) and never saw it fail once, so I suspect that the most efficient type for an "atomic_flag" should be `int` instead of `char`, as was assumed by the original code (which never fails in x86. arm or s390x)

FWIW gcc on this architecture running linux shows:
```
sizeof(atomic_flag) == 1
#define __SIG_ATOMIC_TYPE__ int
#define __SIG_ATOMIC_MAX__ 0x7fffffff
#define __GCC_ATOMIC_INT_LOCK_FREE 2
```